sympy/functions/special/gamma_functions.py

```diff
@@ -701,7 +701,7 @@ def _eval_expand_func(self, **hints):
                         return Add(*tail)/coeff**(n + 1)
                 z *= coeff
 
-        if n == 0 and z.is_rational:
+        if n == 0 and z.is_rational and z.is_Number:
```

Comment:
The last two checks should be one: `z.is_Rational`. Besides being shorter, it's more efficient: it's easier for SymPy to check whether z has that attribute as True (Rational class does), than to calculate whether is_rational can be inferred from known facts about z (when it happens to contain symbols, which will make it fail the last check but not until the second is done).